consul/state/prepared_query.go

```diff
@@ -16,22 +17,54 @@ func isUUID(str string) bool {
 	return validUUID.MatchString(str)
 }
 
+// queryWrapper is an internal structure that is used to store a query alongside
+// its compiled template, which can be nil.
+type queryWrapper struct {
+	*structs.PreparedQuery
```

Comment:
Ah true - I think I originally intended to use it as a query, but I could just put the query in a separate member. I can change that if you'd like.